### PR TITLE
Testing Tic-toc-toe with Rspec

### PR DESCRIPTION
**Pull request by hemantso**
## Testing Tic-toc-toe with Rspec
- input_to_index
- turn_count
- current_player
- position_taken?
- valid_move?
- full?
- over?

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+require './lib/board.rb'
```
Comment (shshamim63) [inline spec/board_spec.rb]: [Optional]
When you need to require a file it is best practice to use `require_relative`.

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './lib/board.rb'
+require './bin/main.rb'
```
Comment (shshamim63) [inline spec/main_spec.rb]: [Optional]
Use require_relative.

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+describe User do
+  let(:game) { Board.new }
+  let(:board) { game.init_board(9) }
+end
```
Comment (shshamim63) [inline spec/main_spec.rb]: I can not see any test cases. Where are the test cases? Well, I checked your main.rb and the class User seems to contain tightly coupled methods. Check out this [article](https://www.codewithjason.com/test-ruby-methods-involve-puts-gets/), and let's find some cases from there that you can add in here.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Required Changes :tractor: 
___
Hi @hemant-soni-vst-au4 :wave: 
The test you added for the board are well organized :clap: an awesome job. But there are some issues that are blocking me to approve this right now. Let's get to them.
___
### :spades: Mandatory Changes :spades: 
___
1. Your readme does not contain any info related to tests. Please add some informative info in there.
2. You have added a test file for the main, but I cannot see any test cases. So please add some.
3. I cannot see any negative test cases in your board_spec. For example, when the valid_mode? should return false?, full? should return false and so on..
___
##### Please fix the above-mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by suretrust — APPROVED** [review]
Hi @hemant-soni-vst-au4, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged. 👍 

_Good luck with your future projects and happy coding!_ 🎉

**Pull request merged**